src/libraries/Pool.sol

```diff
@@ -347,16 +347,17 @@ library Pool {
         StepComputations memory step;
 
         // continue swapping as long as we haven't used the entire input/output and haven't reached the price limit
-        while (state.amountSpecifiedRemaining != 0 && state.sqrtPriceX96 != params.sqrtPriceLimitX96) {
+        while (!(state.amountSpecifiedRemaining == 0 || state.sqrtPriceX96 == params.sqrtPriceLimitX96)) {
             step.sqrtPriceStartX96 = state.sqrtPriceX96;
 
             (step.tickNext, step.initialized) =
                 self.tickBitmap.nextInitializedTickWithinOneWord(state.tick, params.tickSpacing, zeroForOne);
 
             // ensure that we do not overshoot the min/max tick, as the tick bitmap is not aware of these bounds
-            if (step.tickNext < TickMath.MIN_TICK) {
+            if (step.tickNext <= TickMath.MIN_TICK) {
                 step.tickNext = TickMath.MIN_TICK;
-            } else if (step.tickNext > TickMath.MAX_TICK) {
+            }
+            if (step.tickNext >= TickMath.MAX_TICK) {
```

Comment:
why have you removed the `else`?
for the case where `step.tickNext < TickMath.MIN_TICK`, it previously had to go through 1 if check and now it has to go through 2 even though itll never be true for the second one?

Reply:
For most ticks, both checks will be executed. For the boundary, yes there will be one more condition check.
However, for the current test suite, the change resulted in reduction of bytecode size and no change to runtime gas.
https://github.com/Uniswap/v4-core/pull/603/commits/24643eee9033a9f583df225d9583b987bbb19489#diff-a8f8ddb58a9dea35375ee2b5d00cb4f4c3e7c741e05c00655f945609154c6fd1R1